src/basic/path-util.h

```diff
@@ -85,6 +85,19 @@ static inline char* path_simplify(char *path) {
         return path_simplify_full(path, 0);
 }
 
+static inline int path_simplify_alloc(const char *path, char **ret) {
+        char *t = NULL;
+
+        if (path) {
```

Comment:
Maybe, better to shortcut when `NULL` path? To make it clear that this function returns NULL when NULL is passed.
```c
assert(ret);

if (!path) {
        *ret = NULL;
        return 0;
}

char *t = strdup(path);
...
```

Reply:
Updated like this.